packages/tldraw/src/lib/tools/SelectTool/childStates/Idle.ts

```diff
@@ -37,6 +37,9 @@ export class Idle extends StateNode {
 	override onPointerDown: TLEventHandlers['onPointerDown'] = (info) => {
 		if (this.editor.getIsMenuOpen()) return
 
+		// We want to clear the duplicate props when we deselect the current shapes or when we select other shapes
```

Comment:
There are a few ways to deselect shapes, so this won't cover all of them. If we really wanted to be sure, we would need to have the side effect manager clear this part of the state, or have a reactor in the select tool constructor that clears it when the state changes.

In fact, maybe this state should belong to the select tool rather than the global instance state. (There's a bunch of stuff in the instance state that feels kind of tooly rather than instancey, but this could be a first try at a pattern where the state stays with the tool?)

Reply:
Sounds good, will give it a try 👍